src/main/java/it/eng/idsa/businesslogic/service/impl/TLSProvider.java

```diff
@@ -0,0 +1,136 @@
+package it.eng.idsa.businesslogic.service.impl;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.PKIXParameters;
+import java.security.cert.TrustAnchor;
+import java.security.cert.X509Certificate;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.annotation.PostConstruct;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class TLSProvider {
+	
+	private static final Logger logger = LoggerFactory.getLogger(TLSProvider.class);
+
+	private KeyStore tlsKeystore;
+	private KeyStore trustManagerKeyStore;
+	private KeyManagerFactory keyFactory;
+	private TrustManagerFactory trustFactory;
+	
+	@Value("${application.ssl.key-store.name}") String sslKeystore;
+	@Value("${application.ssl.key-store-password}") String sslPassword;
+	@Value("${server.ssl.key-alias}") String sslAlias;
+	@Value("${application.targetDirectory}") Path targetDirectory;
+	@Value("${application.trustStoreName}") String trustStoreName;
+	@Value("${application.trustStorePassword}") String trustStorePwd;
+	
+	@PostConstruct
+	private void loadKeyStores() {
+		loadTLSKeystore();
+		loadTrustStore();
+	}
+
+	private void loadTLSKeystore() {
+		logger.info("Loading key store: " + sslKeystore);
```

Comment:
Maybe we should make this log and on line 71 more distinct, not to cause any confussion if something breaks